### PR TITLE
chore: Adds 4.1.0 RC1 daa to CHANGELOG.md and UPDATING.md

### PR DESCRIPTION
**Pull request by sadpandajoe**
### SUMMARY

This PR:
- Updates `CHANGELOG.md` and `UPDATING.md` with [4.1](https://github.com/apache/superset/tree/4.1) data.
- Updates the version in `superset-front-end/package.json`
- Adds `4.1.0` to version selector of issue template

**NOTE:** Release notes are taken care of in this pr: https://github.com/apache/superset/pull/29262

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by mistercrunch — APPROVED** [review]

**Comment by michael-s-molina** [comment]
Added the `hold` label as this PR can only be merged after the release given that we can include more cherries during the RC cycles.

**Comment by eschutho** [comment]
removing the hold tag and we're going to point the pr to the 4.1 branch instead of master and create a new PR for master later.

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -46,6 +46,7 @@ body:
       label: Superset version
       options:
         - master / latest-dev
+        - "4.1.0"
         - "4.0.2"
```
Comment (michael-s-molina) [inline .github/ISSUE_TEMPLATE/bug-report.yml:50]: You should remove 4.0.2 as 4.1 will be the supported version.

File: superset-frontend/src/dashboard/components/menu/DownloadMenuItems/DownloadScreenshot.test.tsx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+import { logging } from '@superset-ui/core';
+import { DownloadScreenshotFormat } from './types';
+import DownloadScreenshot from './DownloadScreenshot';
+
+const mockAddDangerToast = jest.fn();
+const mockLogEvent = jest.fn();
+const mockAddSuccessToast = jest.fn();
+const mockAddInfoToast = jest.fn();
+
+jest.spyOn(logging, 'error').mockImplementation(() => {});
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/menu/DownloadMenuItems/DownloadScreenshot.test.tsx:33]: Why do we have code updates in this PR? Are we missing some cherries?

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 1 above)
Comment (eschutho) [inline .github/ISSUE_TEMPLATE/bug-report.yml:50]:
```suggestion
```

File: superset-frontend/src/dashboard/components/menu/DownloadMenuItems/DownloadScreenshot.test.tsx (hunk 2 above)
Comment (sadpandajoe) [reply superset-frontend/src/dashboard/components/menu/DownloadMenuItems/DownloadScreenshot.test.tsx]: should be fixed now

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 1 above)
Comment (sadpandajoe) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: I'll do this as part of the master merge branch

**Comment by sadpandajoe** [comment]
@eschutho @michael-s-molina I've followed the changes done in the 4.0.0RC1 branch here: https://github.com/apache/superset/commit/beb9ec77a5d03571bb339c98da31b8e7f9bb5415. I'll create another PR for the official master merge

**Pull request merged**